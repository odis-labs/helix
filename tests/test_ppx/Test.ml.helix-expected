let no_children = Jsx.Elem.div [||] [||]
let single_var_child = Jsx.Elem.div [||] [| v |]
let multi_var_children = Jsx.Elem.div [||] [| v1; v2 |]
let single_lit_child = Jsx.Elem.div [||] [| Jsx.int 42 |]
let mixed_children = Jsx.Elem.div [||] [| v; Jsx.int 42 |]
let empty_frag = Jsx.fragment [||]
let single_var_frag = Jsx.fragment [| v |]
let multi_var_frag = Jsx.fragment [| v1; v2 |]
let single_lit_frag = Jsx.fragment [| Jsx.int 42 |]
let mixed_frag = Jsx.fragment [| v1; Jsx.int 42 |]
let lit_node = Jsx.int 42
