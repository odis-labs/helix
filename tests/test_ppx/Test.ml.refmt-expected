let no_children = (div ~children:[] () [@JSX])
let single_var_child = (div ~children:[ v ] () [@JSX])
let multi_var_children = (div ~children:[ v1; v2 ] () [@JSX])
let single_lit_child = (div ~children:[ 42 ] () [@JSX])
let mixed_children = (div ~children:[ v; 42 ] () [@JSX])
let empty_frag = ([] [@JSX])
let single_var_frag = [ v ] [@JSX]
let multi_var_frag = [ v1; v2 ] [@JSX]
let single_lit_frag = [ 42 ] [@JSX]
let mixed_frag = [ v1; 42 ] [@JSX]
let lit_node = (int ~children:[ 42 ] () [@JSX])
