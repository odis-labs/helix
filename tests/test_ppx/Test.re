// Basic tags
let bt_1 = <t />;
let bt_2 = <t> x </t>;
let bt_3 = <t> x1 x2 </t>;
let bt_4 = <t1> <t2> x </t2> </t1>;
let bt_5 = <t1> <t2> x </t2> <t3 /> </t1>;
let bt_6 = <t1> <t2> x1 </t2> <t3> x2 <t4 /> </t3> </t1>;

// Basic attributes
let ba_1 = <t a=1 />;
let ba_2 = <t a=1> x </t>;
let ba_3 = <t a1=1 a2=2> x </t>;
let ba_4 = <t a />;
let ba_5 = <t a1 a2=2> x </t>;

// Option attributes
let oa_1 = <t ?a />;
let oa_2 = <t a=?x />;

// Option 2-tuple attributes
let o2_1 = <t a=?(s, x) />;

// Option 3-tuple attributes
let o3_1 = <t a=?(s, x, y) />;

// Mixed attributes
let mx_1 = <t a1=1 a2 a3=?x3 ?a4 a5=?(s, 5) a6=?(s, 6, 7)> 0 </t>;

// Inferred type
let it_1 = <t> "hello" </t>;
let it_2 = <t> 5 </t>;
let it_3 = <t> 3.14 </t>;

// Typed tags
let tt_1 = <text> "hello" </text>;
let tt_2 = <int> 42 </int>;
let tt_3 = <float> 3.14 </float>;
let tt_4 = <t> <int> 100 </int> </t>;
