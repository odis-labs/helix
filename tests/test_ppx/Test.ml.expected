let bt_1 = Jsx.Elem.t [||]
let bt_2 = Jsx.Elem.t [||] [| x |]
let bt_3 = Jsx.Elem.t [||] [| x1; x2 |]
let bt_4 = Jsx.Elem.t1 [||] [| Jsx.Elem.t2 [||] [| x |] |]
let bt_5 = Jsx.Elem.t1 [||] [| Jsx.Elem.t2 [||] [| x |]; Jsx.Elem.t3 [||] |]

let bt_6 =
  Jsx.Elem.t1 [||]
    [| Jsx.Elem.t2 [||] [| x1 |]; Jsx.Elem.t3 [||] [| x2; Jsx.Elem.t4 [||] |] |]

let ba_1 = Jsx.Elem.t [| Jsx.Attr.a 1 |]
let ba_2 = Jsx.Elem.t [| Jsx.Attr.a 1 |] [| x |]
let ba_3 = Jsx.Elem.t [| Jsx.Attr.a1 1; Jsx.Attr.a2 2 |] [| x |]
let ba_4 = Jsx.Elem.t [| Jsx.Attr.a a |]
let ba_5 = Jsx.Elem.t [| Jsx.Attr.a1 a1; Jsx.Attr.a2 2 |] [| x |]
let oa_1 = Jsx.Elem.t [| Jsx.Attr.option1 Jsx.Attr.a a |]
let oa_2 = Jsx.Elem.t [| Jsx.Attr.option1 Jsx.Attr.a x |]
let o2_1 = Jsx.Elem.t [| Jsx.Attr.option2 Jsx.Attr.a s x |]
let o3_1 = Jsx.Elem.t [| Jsx.Attr.option3 Jsx.Attr.a s x y |]

let mx_1 =
  Jsx.Elem.t
    [| Jsx.Attr.a1 1
     ; Jsx.Attr.a2 a2
     ; Jsx.Attr.option1 Jsx.Attr.a3 x3
     ; Jsx.Attr.option1 Jsx.Attr.a4 a4
     ; Jsx.Attr.option2 Jsx.Attr.a5 s 5
     ; Jsx.Attr.option3 Jsx.Attr.a6 s 6 7
    |]
    [| Jsx.int 0 |]

let it_1 =
  Jsx.Elem.t [||] [| Jsx.text ("hello" [@reason.raw_literal "hello"]) |]

let it_2 = Jsx.Elem.t [||] [| Jsx.int 5 |]
let it_3 = Jsx.Elem.t [||] [| Jsx.float 3.14 |]
let tt_1 = Jsx.text ("hello" [@reason.raw_literal "hello"])
let tt_2 = Jsx.int 42
let tt_3 = Jsx.float 3.14
let tt_4 = Jsx.Elem.t [||] [| Jsx.int 100 |]
